apm-agent-plugins/apm-ratpack-plugin/src/main/java/co/elastic/apm/agent/ratpack/AbstractExecStarterInstrumentation.java

```diff
@@ -0,0 +1,288 @@
+import co.elastic.apm.agent.web.WebConfiguration;
+import io.netty.handler.codec.http.cookie.Cookie;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import net.bytebuddy.matcher.ElementMatchers;
+import ratpack.exec.ExecStarter;
+import ratpack.exec.Execution;
+import ratpack.func.Action;
+import ratpack.handling.Context;
+import ratpack.registry.RegistrySpec;
+import ratpack.server.ServerConfig;
+
+import java.util.function.Consumer;
+
+import static net.bytebuddy.matcher.ElementMatchers.hasSuperType;
+import static net.bytebuddy.matcher.ElementMatchers.isInterface;
+import static net.bytebuddy.matcher.ElementMatchers.isPublic;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.not;
+import static net.bytebuddy.matcher.ElementMatchers.returns;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+
+public abstract class AbstractExecStarterInstrumentation extends AbstractRatpackInstrumentation {
+
+    private final ElementMatcher<? super MethodDescription> methodMatcher;
+
+    public AbstractExecStarterInstrumentation(ElementMatcher<? super MethodDescription> methodMatcher) {
+        this.methodMatcher = methodMatcher;
+    }
+
+    @Override
+    public final ElementMatcher<? super TypeDescription> getTypeMatcher() {
+        return not(isInterface())
+            .and(hasSuperType(ElementMatchers.named("ratpack.exec.ExecStarter")));
+    }
+
+    @Override
+    public final ElementMatcher<? super MethodDescription> getMethodMatcher() {
+        return methodMatcher;
+    }
+
+    public static class RatpackRegisterTransactionInstrumentation extends AbstractExecStarterInstrumentation {
+
+        public RatpackRegisterTransactionInstrumentation() {
+            super(named("register")
+                .and(isPublic())
+                .and(takesArgument(0, Action.class))
+                .and(returns(ExecStarter.class)));
+        }
+
+        @Advice.OnMethodEnter(suppress = Throwable.class)
+        public static void onBeforeRegister(
+            @Advice.Argument(value = 0, readOnly = false) Action<? super RegistrySpec> action) {
+
+            if (tracer == null) {
+                return;
+            }
+
+            Transaction transaction;
+
+            CoreConfiguration coreConfiguration = tracer.getConfig(CoreConfiguration.class);
+
+            //noinspection ConstantConditions
+            if (coreConfiguration != null
+                && coreConfiguration.isActive()
+                && tracer.currentTransaction() == null) {
+
+
+                // Enhancement: check for excluded urls
+                // @see co.elastic.apm.agent.servlet.ServletTransactionHelper.isExcluded
+
+                transaction = tracer.startTransaction(TraceContext.fromTraceparentHeader(), null, null);
```

Comment:
The second argument should be the `elastic-apm-traceparent` header. The third argument would be the class loader of the class which initiates the transaction. You can inject the current class via `@Advice.This Class<?> clazz` and call `clazz.getClassLoader()` to get the application class loader.